### PR TITLE
Touchup some message documentation and add READMEs with feature docum…

### PR DESCRIPTION
**Pull request by brawner**
This does some minor touchup of the message documentation, but mostly adds READMEs with the feature descriptions of each package.

Signed-off-by: Stephen Brawner <brawner@gmail.com>

### CODE REVIEW

File: sensor_msgs/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# sensor_msgs
+
+This package provides many messages and services relating to sensor devices.
+
+Many of these messages were ported from ROS 1 and a lot of still-relevant documentation can be found through the [ROS 1 sensor_msgs wiki](http://wiki.ros.org/sensor_msgs?distro=noetic).
+
+For more information about ROS 2 interfaces, see [index.ros2.org](https://index.ros.org/doc/ros2/Concepts/About-ROS-Interfaces/)
+
+## sensor_msgs c++ API
+This package provides some common c++ functionality relating to manipulating a couple of particular sensor_msgs messages.
```
Comment (ahcorde) [inline sensor_msgs/README.md:10]:
```suggestion
This package provides some common C++ functionality relating to manipulating a couple of particular sensor_msgs messages.
```

File: sensor_msgs/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+* [Joy](Joy.msg): Reports the state of a joystick's axes and buttons.
+* [LaserEcho](LaserEcho.msg): A submessage of MultiEchoLaserScan and is not intended to be used separately.
+* [LaserScan](LaserScan.msg): Single scan from a planar laser range-finder.
+* [MagneticField](MagneticField.msg): Measurement of the Magnetic Field vector at a specific location.
+* [MultiDOFJointState](MultiDOFJointState.msg): Representation of state for joints with multiple degrees of freedom, following the structure of JointState.
+* [MultiEchoLaserScan](MultiEchoLaserScan.msg): Single scan from a multi-echo planar laser range-finder.
+* [NavSatFix](NavSatFix.msg): Navigation Satellite fix for any Global Navigation Satellite System.
+* [NavSatStatus](NavSatStatus.msg): Navigation Satellite fix status for any Global Navigation Satellite System.
+* [PointCloud2](PointCloud2.msg): Holds a collection of N-dimensional points, which may contain additional information such as normals, intensity, etc.
+* [PointCloud](PointCloud.msg): ** ## THIS MESSAGE IS DEPRECATED AS OF FOXY, use PointCloud2 instead**
```
Comment (ahcorde) [inline sensor_msgs/README.md:39]: Use the same deprecation message here and above

File: geometry_msgs/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# geometry_msgs
+
+This package provides messages for common geometric primitives such as points, vectors, and poses. These primitives are designed to provide a common data type and facilitate interoperability throughout the system.
+
+For more information about ROS 2 interfaces, see [index.ros2.org](https://index.ros.org/doc/ros2/Concepts/About-ROS-Interfaces/)
+
+## Messages (.msg)
+* [Accel](Accel.msg): Expresses acceleration in free space broken into its linear and angular parts.
```
Comment (ahcorde) [inline geometry_msgs/README.md:8]: links are wrong 
```suggestion
* [Accel](msg/Accel.msg): Expresses acceleration in free space broken into its linear and angular parts.
```

File: nav_msgs/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# nav_msgs
+
+This package provides several messages and services for robotic navigation.
+
+For more information about the navigation2 stack in ROS 2, see https://ros-planning.github.io/navigation2/.
+
+For more information about ROS 2 interfaces, see [index.ros2.org](https://index.ros.org/doc/ros2/Concepts/About-ROS-Interfaces/)
+
+## Messages (.msg)
+* [GridCells](GridCells.msg): An array of cells in a 2D grid.
```
Comment (ahcorde) [inline nav_msgs/README.md:10]: links are wrong
```suggestion
* [GridCells](msg/GridCells.msg): An array of cells in a 2D grid.
```

File: nav_msgs/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+* [MapMetaData](MapMetaData.msg): Basic information about the characteristics of the OccupancyGrid.
+* [OccupancyGrid](OccupancyGrid.msg): Represents a 2-D grid map, in which each cell represents the probability of occupancy.
+* [Odometry](Odometry.msg): This represents an estimate of a position and velocity in free space.
+* [Path](Path.msg): An array of poses that represents a Path for a robot to follow.
+
+## Services (.srv)
+* [GetMap](GetMap.srv): Get the map as a nav_msgs/OccupancyGrid.
```
Comment (ahcorde) [inline nav_msgs/README.md:17]:
```suggestion
* [GetMap](srv/GetMap.srv): Get the map as a nav_msgs/OccupancyGrid.
```

File: shape_msgs/README.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# shape_msgs
+
+This package provides several messages and services for describing 3-dimensional shapes.
+
+For more information about ROS 2 interfaces, see [index.ros2.org](https://index.ros.org/doc/ros2/Concepts/About-ROS-Interfaces/)
+
+## Messages (.msg)
+* [Mesh](Mesh.msg): Holds information describing a mesh for visualization and collision detections.
```
Comment (ahcorde) [inline shape_msgs/README.md:8]:
```suggestion
* [Mesh](msg/Mesh.msg): Holds information describing a mesh for visualization and collision detections.
```

Comment (brawner) [reply shape_msgs/README.md]: Fixed.

File: nav_msgs/README.md (hunk 5 above)
Comment (brawner) [reply nav_msgs/README.md]: Fixed.

File: nav_msgs/README.md (hunk 4 above)
Comment (brawner) [reply nav_msgs/README.md]: Fixed.

File: geometry_msgs/README.md (hunk 3 above)
Comment (brawner) [reply geometry_msgs/README.md]: Fixed links in all READMEs

File: sensor_msgs/README.md (hunk 2 above)
Comment (brawner) [reply sensor_msgs/README.md]: Done

File: sensor_msgs/README.md (hunk 1 above)
Comment (brawner) [reply sensor_msgs/README.md]: Done.

File: sensor_msgs/README.md [hunk 7]
```diff
@@ -1,4 +1,4 @@
-# sensor_msgs
+](msg/# sensor_msgs
```
Comment (ahcorde) [inline sensor_msgs/README.md:1]: ?

Comment (ahcorde) [inline sensor_msgs/README.md:1]:
```suggestion
# sensor_msgs
```

**Comment by brawner** [comment]
Err sorry. These commits were already include in #109

**Pull request closed without merging**